src/BlazorWebFormsComponents.Test/LoginControls/Login/Authenticate.razor

```diff
@@ -0,0 +1,58 @@
+﻿@inherits TestComponentBase
+@using System.Security.Claims;
+@using Microsoft.AspNetCore.Components.Authorization
+@using BlazorWebFormsComponents.LoginControls;
+@using Moq;
+
+<Fixture Test="AuthenticateTest">
+	<ComponentUnderTest>
+		<BlazorWebFormsComponents.LoginControls.Login @ref="loginControl" ID="loginControl" OnAuthenticate="Login_Authenticate" />
+	</ComponentUnderTest>
+</Fixture>
+
+@code {
+	Login loginControl = new Login();
+	bool isAuthenticated = false;
+
+	void AuthenticateTest()
+	{
+		var principal = new ClaimsPrincipal();
+		var identity = new ClaimsIdentity(new Claim[] { new Claim(ClaimTypes.Name, "James Bond") }, "Test authentication");
+		principal.AddIdentity(identity);
+
+		var authenticationStateProviderMock = new Mock<AuthenticationStateProvider>();
+		authenticationStateProviderMock.Setup(x => x.GetAuthenticationStateAsync()).Returns(Task.FromResult(new AuthenticationState(principal)));
+
+		Services.AddSingleton<AuthenticationStateProvider>(authenticationStateProviderMock.Object);
+
+		var navigationManagerMock = new Mock<NavigationManager>();
+		Services.AddSingleton<NavigationManager>(navigationManagerMock.Object);
+
+		var cut = GetComponentUnderTest();
+
+		cut.Find("#loginControl_UserName").Change("admin");
+		cut.Find("#loginControl_Password").Change("Fake Password");
+		cut.Find("form").Submit();
+		cut.Markup.ShouldContain("Your login attempt was not successful");
+		isAuthenticated.ShouldBeFalse();
+
+		cut.Find("#loginControl_UserName").Change("admin");
+		cut.Find("#loginControl_Password").Change("SecRet P@ssw0rd");
+		cut.Find("form").Submit();
+		cut.Markup.ShouldNotContain("Your login attempt was not successful");
+		isAuthenticated.ShouldBeTrue();
+	}
+
+	void Login_Authenticate(AuthenticateEventArgs args)
+	{
+		if (loginControl.UserName == "admin" && loginControl.Password == "SecRet P@ssw0rd")
```

Comment:
Can we replace these strings with a pair of constants so that our test methods use those and are guaranteed that they match?

We want to reduce the number of 'magic strings' in our code